### PR TITLE
Move the componentName to the front of the method

### PR DESCRIPTION
**Pull request by jamdavi**

### CODE REVIEW

File: shared/src/ClientPropertyCollection.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ public class ClientPropertyCollection : PayloadCollection
     {
         private const string VersionName = "$version";
 
-        /// <inheritdoc path="/exception['ArgumentException']" cref="Add(IDictionary{string, object}, string, bool)" />
+        /// <inheritdoc path="/exception['ArgumentException']" cref="AddInternal(IDictionary{string, object}, string, bool)" />
```
Comment (jamdavi) [inline shared/src/ClientPropertyCollection.cs:17]: It was getting a bit confusing with all of the Adds

File: shared/src/ClientPropertyCollection.cs [hunk 2]
```diff
@@ -158,7 +196,7 @@ public void AddOrUpdate(string propertyName, object propertyValue, int statusCod
         /// <param name="properties">A collection of properties to add or update.</param>
         /// <param name="componentName">The component with the properties to add or update.</param>
         /// <param name="forceUpdate">Forces the collection to use the Add or Update behavior. Setting to true will simply overwrite the value; setting to false will use <see cref="IDictionary{TKey, TValue}.Add(TKey, TValue)"/></param>
-        private void Add(IDictionary<string, object> properties, string componentName = default, bool forceUpdate = false)
+        private void AddInternal(IDictionary<string, object> properties, string componentName = default, bool forceUpdate = false)
```
Comment (jamdavi) [inline shared/src/ClientPropertyCollection.cs:212]: It was getting a bit confusing with all of the Adds

File: shared/src/ClientPropertyCollection.cs [hunk 3]
```diff
@@ -73,44 +97,42 @@ public void Add(IDictionary<string, object> properties, string componentName = d
         /// <param name="version"></param>
         /// <param name="description"></param>
         /// <param name="componentName"></param>
-        public void Add(string propertyName, object propertyValue, int statusCode, long version, string description = default, string componentName = default)
+        public void Add(string componentName, string propertyName, object propertyValue, int statusCode, long version, string description = default)
         {
             if (Convention?.PayloadSerializer == null)
```
Comment (abhipsaMisra) [inline shared/src/ClientPropertyCollection.cs:102]: q = would this ever be null, we'd always default this to newtonsoft, right?

Comment (abhipsaMisra) [reply shared/src/ClientPropertyCollection.cs:102]: actually nvm, this is right -> null when created from scratch and value available if reusing from a returned property.

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**